### PR TITLE
feat: Release notes for Codacy Self-hosted 10.0.0 DOCS-474

### PR DESCRIPTION
**Pull request by machadoit**
Adds auto-generated release notes for Codacy Self-hosted 10.0.0

### :eyes: Live preview
https://feat-release-notes-self-hosted-10-0--docs-codacy.netlify.app/release-notes/self-hosted/self-hosted-v10.0.0/

### 🚧 To do
- [x] Update the variable `extra.version` in `mkdocs.yml` to the new Codacy Self-hosted version
- [x] Add the new release notes page to `mkdocs.yml` and `docs/release-notes/index.md`
- [x] Review generated release notes
- [x] Review list of issues with missing release notes
- [x] Ensure that links point to documentation for new Codacy Self-hosted version
- [x] Review links to changelogs of updated tools
- [x] Add breaking changes
- [x] Include the release note from [COV-25](https://codacy.atlassian.net/browse/COV-25) (see [COV-108](https://codacy.atlassian.net/browse/COV-108))
- [x] Add the [correct CLI versions](https://github.com/codacy/docs/pull/1517#discussion_r1034903602)
- [x] Merge https://github.com/codacy/docs/pull/1525
- [x] Update release date and remove `TODO` comments
- [x] Deploy the version of the docs already prepared on the release branch [release/v10.0](https://github.com/codacy/docs/tree/release/v10.0)

[COV-108]: https://codacy.atlassian.net/browse/COV-108?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[COV-25]: https://codacy.atlassian.net/browse/COV-25?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by machadoit — COMMENTED** [review]
LGTM! :D

File: docs/release-notes/self-hosted/self-hosted-v10.0.0.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Follow the steps below to upgrade to Codacy Self-hosted v10.0.0:
+
+1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v10.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the versions with the Git tag `self-hosted-10.0.0`:
+
+    -   [Codacy Analysis CLI](https://github.com/codacy/codacy-analysis-cli/releases/tag/self-hosted-10.0.0)
+    -   [Codacy Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter/releases/tag/self-hosted-10.0.0)
```
Comment (prcr) [inline docs/release-notes/self-hosted/self-hosted-v10.0.0.md:26]: **To do:** Check https://github.com/codacy/docs/pull/1528 and apply similar changes here

File: docs/release-notes/self-hosted/self-hosted-v10.0.0.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+Follow the steps below to upgrade to Codacy Self-hosted v10.0.0:
+
+1.  Check the [release notes for all Codacy Self-hosted versions](../index.md#self-hosted) **between your current version and the most recent version** for breaking changes and follow the instructions provided <span class="skip-vale">carefully</span>.
+
+1.  Follow the instructions to [upgrade your Codacy Self-hosted instance](https://docs.codacy.com/v10.0/chart/maintenance/upgrade/).
+
+1.  Update your Codacy command-line tools to the following versions:
+
+    -   [Codacy Analysis CLI 7.6.6](https://github.com/codacy/codacy-analysis-cli/releases/tag/7.6.6)
+    -   [Codacy Coverage Reporter 13.10.15](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.10.15)
```
Comment (prcr) [inline docs/release-notes/self-hosted/self-hosted-v10.0.0.md:26]: Validate that these are the correct versions (this was copied from the [Codacy Self-hosted 9.0.0 release notes](https://docs.codacy.com/release-notes/self-hosted/self-hosted-v9.0.0/#upgrading-codacy-self-hosted)).

Comment (lventura-codacy) [reply docs/release-notes/self-hosted/self-hosted-v10.0.0.md:26]: We have created selfhosted tags ([self-hosted-10.0.0](https://github.com/codacy/codacy-analysis-cli/releases/tag/self-hosted-10.0.0)) which match the latest versions. Can we go with those? If we want the specific cli tool versions they are:
[Codacy Analysis CLI 7.7.4](https://github.com/codacy/codacy-analysis-cli/releases/tag/7.7.4)
[Codacy Coverage Reporter 13.10.15](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.11.17)

Comment (nicklem) [reply docs/release-notes/self-hosted/self-hosted-v10.0.0.md:26]: If we want to revert to the tag links:

```suggestion
1.  Update your Codacy command-line tools to the  versions with the Git tag `self-hosted-10.0.0`:

    -   [Codacy Analysis CLI](https://github.com/codacy/codacy-analysis-cli/releases/tag/self-hosted-10.0.0)
    -   [Codacy Coverage Reporter](https://github.com/codacy/codacy-coverage-reporter/releases/tag/self-hosted-10.0.0)
```

Comment (nicklem) [reply docs/release-notes/self-hosted/self-hosted-v10.0.0.md:26]: If we want to stay with the [same versioning syntax we use in SH 9](https://docs.codacy.com/release-notes/self-hosted/self-hosted-v9.0.0/#upgrading-codacy-self-hosted):

```suggestion
1.  Update your Codacy command-line tools to the following versions:

    -   [Codacy Analysis CLI 7.7.4](https://github.com/codacy/codacy-analysis-cli/releases/tag/7.7.4)
    -   [Codacy Coverage Reporter 13.10.15](https://github.com/codacy/codacy-coverage-reporter/releases/tag/13.10.15)
```

**Review by lventura-codacy — APPROVED** [review]
LGTM!

**Comment by nicklem** [comment]
Reverted tool version updates to what Paulo suggests, given the "[general consensus](https://codacy.slack.com/archives/CSSSHAK9N/p1675433164911829?thread_ts=1675261879.083599&cid=CSSSHAK9N)" 😄

**Pull request merged**